### PR TITLE
Enhancement for DateValidator (#7177, #10165):

### PR DESCRIPTION
**Pull request by VirtualRJ**
- Added parameter to define if checking for 'date' or 'dateTime' and updated parseDateValueIntl($value, $format) to facilitate this validation based on defined type.
- Added unit tests to for validating dateTime values when using short format validation.

### CODE REVIEW

**Comment by VirtualRJ** [comment]
@cebe I have not added a line to changelog since there was no section for 2.0.8 yet and not sure which version this will be included (if at all).

Just so this pull request will be referenced in the issues: (#7177, #10165)

File: framework/validators/DateValidator.php [hunk 1]
```diff
@@ -143,7 +143,15 @@ class DateValidator extends Validator
...
     public $minString;
-
+    /**
+     * @var string type used for defining if short format validation should be on just
+     * date or on datetime. Defaults to 'date'. Possible values are:
+     * - `date`
+     * - `dateTime`
+     * @since 2.0.8
+     */
+    public $type = 'date';
```
Comment (SilverFire) [inline framework/validators/DateValidator.php]: I'm suggesting to introduce constants for types

**Comment by SilverFire** [comment]
We need a changelog line

**Comment by SilverFire** [comment]
What about JS?

**Comment by cebe** [comment]
> What about JS client-side validation?

DateValidator does not have client validation as it would be too complex to duplicate the functionality in JS.

**Comment by SilverFire** [comment]
Ok

**Comment by VirtualRJ** [comment]
Added the suggested constants.

@SilverFire, @cebe: Where should I add the line for the changelog? As mentioned earlier there is no section for 2.0.8 (according to the issue this is added to the 2.0.8 milestone) and I believe 2.0.7 is frozen for release, right?

**Comment by samdark** [comment]
@VirtualRJ you're right. You may post changelog line right here in comments, we'll handle it during merge.

**Comment by VirtualRJ** [comment]
Ok! Changelog line would be:

```
- Enh #7177, #10165: Added support for validating datetime using short format (VirtualRJ)
```

**Comment by cebe** [comment]
merged, with a few adjustments: 2ac17dd..00d506c Thank you!

**Pull request closed without merging**